### DISCUSSION TITLE
Inconsistent names resolution

### DISCUSSION THREAD

**Discussion by vitvakatu** [enso-org/enso#6026] [Work In Progress - Engine]
Hi! I'm developing recognition of qualified names in CB and faced a weird behavior of the engine.

I have a project named `Unnamed_14` with the submodule `Foo` with the method `foo`. I also have `func1` defined in the `Main` module of the project.

1. For `func1`, I have expected results – I can call it by the fully qualified name `local.Unnamed_14.func1`, and I can import any part of the name and make it shorter.
2. `local.Unnamed_14.Main.func1` does not work though. It seems inconsistent, as I can call `Main.func1`. It fails with `method Main of Main could not be found`.
3. I can't call `foo` by a fully qualified name. It works with `func1`, so it is inconsistent. It fails with `method Foo of Main could not be found`.

I wonder what the correct behavior in such cases is. Do we have any bugs here, or it is all working as designed?

<img width="868" alt="Screenshot 2023-03-21 at 14 10 07" src="https://user-images.githubusercontent.com/6566674/226575817-2e4135b2-8df7-403f-9259-fccf691c5a92.png">

[Unnamed_14.zip](https://github.com/enso-org/enso/files/11027146/Unnamed_14.zip)

**Comment by farmaazon** [comment]
I made an issue #5616 AFAIK the "Main" name should not be used in code at all.
